frontend/public/components/modals/column-management-modal.tsx

```diff
@@ -0,0 +1,170 @@
+          aria-labelledby="table-column-management-item1"
+          checked={column.visible}
+          name={column.title}
+          onChange={onChange}
+        />
+        <DataListItemCells
+          dataListCells={[
+            <DataListCell id={`table-column-management-item-${column.title}`} key={column.title}>
+              {column.title}
+            </DataListCell>,
+          ]}
+        />
+      </DataListItemRow>
+    </DataListItem>
+  ));
+
+export const ColumnManagementModal: React.FC<ColumnManagementModalProps> = ({
+  kinds,
+  cancel,
+  close,
+  columnFilters,
+}) => {
+  const dispatch = useDispatch();
+  const initialDefaultColumns = !_.isEmpty(columnFilters)
+    ? _.cloneDeep(columnFilters).filter((column) => column.title.length > 0 && !column.additional)
+    : {};
+  const initialAdditionalColumns = !_.isEmpty(columnFilters)
+    ? _.cloneDeep(columnFilters).filter((column) => column.additional)
+    : {};
+  const [defaultColumns, setDefaultColumns] = React.useState<any[]>(initialDefaultColumns);
+
+  const [additionalColumns, setAdditionalColumns] = React.useState<ColumnHeaderRow[]>(
+    initialAdditionalColumns,
+  );
+
+  const updateColumns = (checked: boolean, name: string, columnArray: any): any => {
+    const updatedColumns = _.cloneDeep(columnArray);
+    return updatedColumns.map((column) => {
+      if (_.isEmpty(name) || column.title === name) {
+        column.visible = checked;
+        return column;
+      }
+      return column;
+    });
+  };
+
+  const defaultColumnChange = (checked: boolean, event): void => {
+    const target = event.target;
+    setDefaultColumns(updateColumns(checked, target.name, defaultColumns));
+  };
+
+  const additionalColumnChange = (checked: boolean, event): void => {
+    const target = event.target;
+    setAdditionalColumns(updateColumns(checked, target.name, additionalColumns));
+  };
+
+  const submit = (event): void => {
+    event.preventDefault();
+    dispatch(setColumnManagementFilter(kinds[0], [...defaultColumns, ...additionalColumns]));
+    close();
+  };
+
+  const areMaxColumnsDisplayed =
+    additionalColumns.filter((column) => column.visible).length +
+      defaultColumns.filter((column) => column.visible).length ===
+    MAX_VIEW_COLS;
+
+  const resetColumns = (event): void => {
+    event.preventDefault();
+    setDefaultColumns(updateColumns(true, null, defaultColumns));
+    setAdditionalColumns(updateColumns(false, null, additionalColumns));
+  };
+
+  return (
+    <form onSubmit={submit} name="form" className="modal-content">
+      <ModalTitle className="modal-header">Manage Columns</ModalTitle>
+      <ModalBody>
+        <div className="co-m-form-row">
+          <p>Checked categories will be displayed in the table.</p>
+        </div>
+        <div className="row co-m-form-row">
+          <Alert
+            className="co-alert"
+            isInline
+            title={`You can select up to ${MAX_VIEW_COLS - 1} columns`}
```

Comment:
I assume we're subtracting one here because name is disabled? Will that be confusing because we show name as selected in the dialog?

Reply:
removing the subtraction i think that might be more clear.